fact-ebpf/src/bpf/main.c

```diff
@@ -104,3 +105,91 @@ int BPF_PROG(trace_file_open, struct file* file) {
   m->file_open.ignored++;
   return 0;
 }
+
+SEC("lsm/path_unlink")
+int BPF_PROG(trace_path_unlink, struct path* dir, struct dentry* dentry) {
+  uint32_t key = 0;
+  struct event_t* event = NULL;
+  struct metrics_t* m = bpf_map_lookup_elem(&metrics, &key);
+  if (m == NULL) {
+    bpf_printk("Failed to get metrics entry, this should not happen");
+    return 0;
+  }
+
+  m->path_unlink.total++;
+
+  struct path_cfg_helper_t* prefix_helper = bpf_map_lookup_elem(&path_prefix_helper, &key);
+  if (prefix_helper == NULL) {
+    bpf_printk("Failed to get prefix helper");
+    goto error;
+  }
+
+  long path_len = bpf_d_path(dir, prefix_helper->path, PATH_MAX);
+  if (path_len <= 0 || path_len > 4096) {
+    bpf_printk("Failed to read path");
+    goto error;
+  }
+  prefix_helper->path[path_len - 1] = '/';
+
+  struct qstr d_name;
+  BPF_CORE_READ_INTO(&d_name, dentry, d_name);
+  int len = d_name.len;
+  if (len + path_len > 4096) {
+    bpf_printk("Invalid path length: %u", len + path_len);
+    goto error;
+  }
+
+  if (bpf_probe_read_kernel(&prefix_helper->path[path_len & 0xFFF], len & 0xFFF, d_name.name)) {
```

Comment:
Can't we just use `path_len` and `len` directly since we know that they're both <4096 already?